apis/bigquerybiglake/v1alpha1/table_identity.go

```diff
@@ -69,6 +71,38 @@ func NewTableIdentity(ctx context.Context, reader client.Reader, obj *BigLakeTab
 		return nil, fmt.Errorf("cannot resolve location")
 	}
 
+	catalogRef := obj.Spec.CatalogRef
+	if catalogRef == nil {
+		return nil, fmt.Errorf("cannot resolve catalog")
+	}
```

Comment:
Suggest making this a API level validation  ("required") rather than controller level https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/api-conventions/validations.md#basic-rules

Reply:
This is required today!

https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/ad92d08fa0661a42382f9508596076249230fcc5/apis/bigquerybiglake/v1alpha1/table_types.go#L32-L34

But I always get a little defensive when coding and handling references. Do we want to just rely on the CRD validation?

Reply:
We handle reference in controller (Adaptor initialization for required field, and Adaptor Create/Update for optional). 

But my previous comment is for required/optional, and not for reference.  Required and optional could be done via CRD validation unless it has additional logic (e.g. 3 fields have a anyOf requirement)